sw/device/silicon_creator/lib/drivers/flash_ctrl.c

```diff
@@ -523,157 +523,52 @@ flash_ctrl_cfg_t flash_ctrl_data_default_cfg_get(void) {
-        mp_region_cfg,                                                                        \
-        FLASH_CTRL_MP_REGION_CFG_##region_macro_arg##_PROG_EN_##region_macro_arg##_FIELD,     \
-        perms.write);                                                                         \
-    mp_region_cfg = bitfield_field32_write(                                                   \
-        mp_region_cfg,                                                                        \
-        FLASH_CTRL_MP_REGION_CFG_##region_macro_arg##_RD_EN_##region_macro_arg##_FIELD,       \
-        perms.read);                                                                          \
-    mp_region_cfg = bitfield_field32_write(                                                   \
-        mp_region_cfg,                                                                        \
-        FLASH_CTRL_MP_REGION_CFG_##region_macro_arg##_EN_##region_macro_arg##_FIELD,          \
-        en);                                                                                  \
-    sec_mmio_write32(                                                                         \
-        kBase + FLASH_CTRL_MP_REGION_CFG_##region_macro_arg##_REG_OFFSET,                     \
-        mp_region_cfg);                                                                       \
-    if (lock != kHardenedBoolFalse) {                                                         \
-      sec_mmio_write32(                                                                       \
-          kBase +                                                                             \
-              FLASH_CTRL_REGION_CFG_REGWEN_##region_macro_arg##_REG_OFFSET,                   \
-          0);                                                                                 \
-    }                                                                                         \
-    return;                                                                                   \
-  }
-
-  switch (launder32(region)) {
-    FLASH_CTRL_MP_REGIONS(FLASH_CTRL_MP_REGION_CFG_WRITE_)
-    default:
-      OT_UNREACHABLE();
-  }
-
-#undef FLASH_CTRL_MP_REGION_CFG_WRITE_
-}
-
 void flash_ctrl_data_region_protect(flash_ctrl_region_index_t region,
                                     uint32_t page_offset, uint32_t num_pages,
                                     flash_ctrl_perms_t perms,
                                     flash_ctrl_cfg_t cfg,
                                     hardened_bool_t lock) {
+  HARDENED_CHECK_GE(region, 0);
+  HARDENED_CHECK_LT(region, 8);
+  region *= sizeof(uint32_t);
+
   // Reset the region's configuration via the MP_REGION_CFG_${region} register.
   // This temporarily disables memory protection for the region.
-  flash_ctrl_mp_region_cfg_reset(region);
+  sec_mmio_write32(kBase + FLASH_CTRL_MP_REGION_CFG_0_REG_OFFSET + region,
```

Comment:
We should consider adding a `SEC_MMIO_ASSERT_WRITE_INCREMENT()` to keep track of the `sec_mmio` counter increment.  

I am not sure if we are doing any sec_mmio checks in the rom_ext after ownership commands have been executed. This is something we can discuss as a separate issue.

Reply:
Sounds good. I think we can do this in a follow up PR once we review the `sec_mmio` hardening for the  ROM_EXT.

Reply:
Done.